### PR TITLE
Support interrupt conditions

### PR DESCRIPTION
**Pull request by AlexanderPrendota**

### CODE REVIEW

**Comment by AlexanderPrendota** [comment]
@wayfarer-rus Let's do improvement in current branch  =)

**Comment by wayfarer-rus** [comment]
removed all unnecessary entities. 

code looks much cleaner like that.

we lost ability to accumulate output buffers if timeout happens though.

File: src/main/kotlin/com/compiler/server/executor/JavaExecutor.kt [hunk 1]
```diff
@@ -2,25 +2,34 @@ package com.compiler.server.executor
...
+    const val MAX_OUTPUT_SIZE = 100 * 1024
+    const val EXECUTION_TIMEOUT = 10000L
+  }
+
   data class ProgramOutput(
     val standardOutput: String,
     val errorOutput: String,
     val exception: Exception? = null
   ) {
+    constructor(exception: Exception) : this("", "", exception)
```
Comment (AlexanderPrendota) [inline src/main/kotlin/com/compiler/server/executor/JavaExecutor.kt]: Why u need the constructor? We can use default values instead

File: src/main/kotlin/com/compiler/server/executor/JavaExecutor.kt [hunk 2]
```diff
@@ -2,25 +2,34 @@ package com.compiler.server.executor
...
+
     fun asExecutionResult(): JavaExecutionResult {
       return JavaExecutionResult(
-        text = "<outStream>$standardOutput\n</outStream>",
+        text = "<outStream>$standardOutput\n</outStream><errStream>$errorOutput</errStream>",
```
Comment (AlexanderPrendota) [inline src/main/kotlin/com/compiler/server/executor/JavaExecutor.kt]: Hm? error after standard?
```
System.out("1")
System.err("123")
System.out("2")
```
probably the output will be
1
2
123

does it correct?)

File: src/main/kotlin/com/compiler/server/executor/JavaExecutor.kt [hunk 3]
```diff
@@ -29,72 +38,94 @@ class JavaExecutor {
 
   fun execute(args: List<String>): JavaExecutionResult {
     return Runtime.getRuntime().exec(args.toTypedArray()).use {
+      // why?
```
Comment (AlexanderPrendota) [inline src/main/kotlin/com/compiler/server/executor/JavaExecutor.kt]: Drop why

File: src/main/kotlin/com/compiler/server/executor/JavaExecutor.kt (hunk 2 above)
Comment (wayfarer-rus) [reply src/main/kotlin/com/compiler/server/executor/JavaExecutor.kt]: No, it is simpler.

If you have tag <outStream>, why not to have also <errStream>?

You should think about using it somehow, though

File: src/main/kotlin/com/compiler/server/executor/JavaExecutor.kt (hunk 1 above)
Comment (wayfarer-rus) [reply src/main/kotlin/com/compiler/server/executor/JavaExecutor.kt]: Good idea!

I write too much Java code lately.

**Pull request merged**